lib/MusicBrainz/Server/Controller/Role/CommonsImage.pm

```diff
@@ -13,16 +13,22 @@ Readonly my $WIKIDATA_PROP_LOCATOR_MAP_IMAGE => "P242";
 after load => sub {
     my ($self, $c) = @_;
 
-    $self->_get_commons_image($c, 1);
+    my $entity = $c->stash->{entity};
+    my $endpoint = $c->uri_for_action($self->action_for('commons_image'), [$entity->gid]);
+    $c->stash->{image} = $self->_get_commons_image($c, 1);
```

Comment:
This runs on every page load. The `1` passed as the second argument to `_get_commons_image` indicates that it should only try loading the image info from the cache. If it's not cached, it's loaded asynchronously after the page loads. (This is not new behavior.)